deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/NeuralNetConfiguration.java

```diff
@@ -1008,17 +1008,23 @@ public Builder l2Bias(double l2Bias) {
         }
 
         /**
-         * Dropout probability. This is the probability of <it>retaining</it> an activation. So dropOut(x) will keep an
-         * activation with probability x, and set to 0 with probability 1-x.<br>
-         * dropOut(0.0) is disabled (default).
+         * Dropout probability. This is the probability of <it>retaining</it> an input activation for a layer. So
+         * dropOut(x) will keep an input activation with probability x, and set to 0 with probability 1-x.<br>
```

Comment:
I am not sure whether I understand this. Because if I do, I think it is wrong.
Dropout should "zero out _nodes_", and I've come to understand that this means that their OUTPUT will be zero. Which again means that the NEXT layer will see a consistent set of zeroed and non-zeroed node values, i.e. if for Layer L node 0, 2 and 5 is zeroed, then ALL nodes in Layer L+1 will see node 0, 2 and 5 as zeroed.

Reading this JavaDoc, I get the feeling that node 0 in Layer L+1 (which is THIS layer, since we're talking "input activations") might see that the input activation is zeroed for node 1, 2, 3 from Layer L, while node 1 in Layer L+1 could see zeroed nodes for 0, 2 and 3 - i.e. independently zeroed.

If the latter is the case, then I believe this dropout is implemented incorrectly. But I probably misunderstand the wording.

Also, with dropout, one should have this scaling thing. This should be mentioned in the JavaDoc. Here's TensorFlow's dropout doc:
https://www.tensorflow.org/api_docs/python/tf/nn/dropout

"With probability keep_prob, outputs the input element scaled up by 1 / keep_prob, otherwise outputs 0. The scaling is so that the expected sum is unchanged."

Which brings us to the next thing that should be javadocced, and that is that dropout should only be applied in the training phase, not the evaluation. In particular, this also goes for the scaling. Since when using TensorFlow, you as user needs to set the keep_prob to your wanted value when training, while in evaluation mode, **you need to set it to 1**. Using e.g. "0.5", this means that when training, half of the nodes are zeroed, but the output is scaled by 2 for the remaining. But in evaluation, no nodes are silenced, and there is no scaling.

Is this handled automatically by sl4j? Would be nice to have it here in the JavaDoc.

Regarding whether it is "before" or "after" this layer: This could be explicitly explained like "setting a dropout on this layer is the same as adding a {@link DropoutLayer} _in front_ of this layer".